core/src/test/java/org/elasticsearch/index/query/InnerHitBuilderTests.java

```diff
@@ -221,7 +221,12 @@ public static InnerHitBuilder randomInnerHits(boolean recursive, boolean include
         innerHits.setTrackScores(randomBoolean());
         innerHits.setFieldNames(randomListStuff(16, () -> randomAsciiOfLengthBetween(1, 16)));
         innerHits.setFieldDataFields(randomListStuff(16, () -> randomAsciiOfLengthBetween(1, 16)));
-        innerHits.setScriptFields(new HashSet<>(randomListStuff(16, InnerHitBuilderTests::randomScript)));
+        // Random script fields deduped on their field name.
+        Map<String, SearchSourceBuilder.ScriptField> scriptFields = new HashMap<>();
```

Comment:
The equals method checks all the fields I believe.
On May 5, 2016 6:20 PM, "Lee Hinman" <notifications@github.com> wrote:

> In
> core/src/test/java/org/elasticsearch/index/query/InnerHitBuilderTests.java
> <https://github.com/elastic/elasticsearch/pull/18171#discussion_r62264898>
> :
>
> > @@ -221,7 +221,12 @@ public static InnerHitBuilder randomInnerHits(boolean recursive, boolean include
> >          innerHits.setTrackScores(randomBoolean());
> >          innerHits.setFieldNames(randomListStuff(16, () -> randomAsciiOfLengthBetween(1, 16)));
> >          innerHits.setFieldDataFields(randomListStuff(16, () -> randomAsciiOfLengthBetween(1, 16)));
> > -        innerHits.setScriptFields(new HashSet<>(randomListStuff(16, InnerHitBuilderTests::randomScript)));
> > +        // Random script fields deduped on their field name.
> > +        Map<String, SearchSourceBuilder.ScriptField> scriptFields = new HashMap<>();
>
> ScriptField implements .equals(), so perhaps just use a HashSet instead
> of constructing a map and then calling .values() on it?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/18171/files/698ab3928839d525822c6da036786ccd4c77217c#r62264898>
>